typing/ctype.ml

```diff
@@ -1756,8 +1760,20 @@ let enforce_constraints env ty =
 
 (* Recursively expand the head of a type.
    Also expand #-types. *)
-let full_expand env ty =
-  let ty = repr (expand_head env ty) in
+let full_expand ~may_forget_scope env ty =
```

Comment:
I'm not sure that the changes to `full_expand` are the right behaviour. I think that a node whose level is outside the scope of the equation should probably be printed as `type a` rather than `type a = foo` since it would not be expanded without producing ambiguity.

Reply:
If the error came from this expansion, it seems better to show it, otherwise the message is rather confusing.
Maybe I could try to refine this behavior, so that we can explain in the error message that this (probably expected) expansion is not available here.